base/sparse/cholmod.jl

```diff
@@ -860,7 +860,18 @@ convert(::Type{Dense}, A::Sparse) = sparse_to_dense(A)
 function (::Type{Sparse}){Tv<:VTypes}(m::Integer, n::Integer,
         colptr::Vector{SuiteSparse_long}, rowval::Vector{SuiteSparse_long},
         nzval::Vector{Tv}, stype)
-    # check if columns are sorted
+    # checks
+    ## length of input
+    if length(nzval) != length(rowval)
```

Comment:
I can see why you'd like to allow `length(nzval) > colptr[end]` but why would you allow `length(nzval) != length(rowval)`? It also appears that CHOLMOD assumes that these have the same size.